tests/components/seventeentrack/__init__.py

```diff
@@ -1 +1,150 @@
 """Tests for the seventeentrack component."""
+
+import datetime
+from unittest.mock import patch
+
+from py17track.errors import SeventeenTrackError
+
+from homeassistant.components.seventeentrack import DOMAIN
+from homeassistant.components.seventeentrack.const import (
+    CONF_SHOW_ARCHIVED,
+    CONF_SHOW_DELIVERED,
+)
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.util import utcnow
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+VALID_PLATFORM_CONFIG_FULL = {
+    "sensor": {
+        "platform": "seventeentrack",
+        CONF_USERNAME: "test",
+        CONF_PASSWORD: "test",
+        CONF_SHOW_ARCHIVED: True,
+        CONF_SHOW_DELIVERED: True,
+    }
+}
+
+DEFAULT_SUMMARY = {
+    "Not Found": 0,
+    "In Transit": 0,
+    "Expired": 0,
+    "Ready to be Picked Up": 0,
+    "Undelivered": 0,
+    "Delivered": 0,
+    "Returned": 0,
+}
+
+NEW_SUMMARY_DATA = {
+    "Not Found": 1,
+    "In Transit": 1,
+    "Expired": 1,
+    "Ready to be Picked Up": 1,
+    "Undelivered": 1,
+    "Delivered": 1,
+    "Returned": 1,
+}
+
+VALID_CONFIG = {
+    CONF_USERNAME: "test",
+    CONF_PASSWORD: "test",
+}
+
+INVALID_CONFIG = {"notusername": "seventeentrack", "notpassword": "test"}
+
+VALID_OPTIONS = {
+    CONF_SHOW_ARCHIVED: True,
+    CONF_SHOW_DELIVERED: True,
+}
+
+NO_DELIVERED_OPTIONS = {
+    CONF_SHOW_ARCHIVED: False,
+    CONF_SHOW_DELIVERED: False,
+}
+
+
+async def _goto_future(hass, future=None):
+    """Move to future."""
+    if not future:
+        future = utcnow() + datetime.timedelta(minutes=10)
+    with patch("homeassistant.util.utcnow", return_value=future):
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
+
```

Comment:
We use the freezer for these tasks. We can do `freezer.tick(..)` and it will skip time (should be combined with async fire time changed).